### PR TITLE
*: Reduce memory operations by simply passing size hint to `make`

### PR DESCRIPTION
**Pull request by innerr**
I checked the usage of heap memory, and try to reduce memory allocating.
Here are some "innocent" changes to help to do that.
I won't expect too much improvement on this PR, after all most of the changes are not in the hot path.

During the examination I also made some changes to some spots that involved more than one or two lines, but they would not be here, I will make sure they are in the hot path before push them up.

### What problem does this PR solve? 
Reduce memory allocating slightly

### What is changed and how it works?
By passing size hint to `make`

### Check List 

Tests 

 - Integration test

Code changes
 - None

Side effects
 - None

### CODE REVIEW

**Comment by innerr** [comment]
Latency looks slightly improved

**Comment by innerr** [comment]
@lzmhhh123 PTAL, and what other things I should do to get another reviewer?

**Comment by lzmhhh123** [comment]
@innerr You can request others to review your PR by click "request a review".

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -304,7 +304,11 @@ func (hch *handleColHelper) pushMap(m map[int64][]*expression.Column) {
 }
 
 func (hch *handleColHelper) mergeAndPush(m1, m2 map[int64][]*expression.Column) {
-	newMap := make(map[int64][]*expression.Column)
+	mapMinSize := len(m1)
+	if len(m2) > len(m1) {
```
Comment (coocood) [inline planner/core/planbuilder.go]: We can use `mathutil.Min` function to simplify the code.

File: sessionctx/stmtctx/stmtctx.go [hunk 2]
```diff
@@ -548,7 +548,11 @@ func (sc *StatementContext) CopTasksDetails() *CopTasksDetails {
 		sleepTime time.Duration
 		times     int
 	}
-	backoffInfo := make(map[string][]backoffItem)
+	backoffNum := 0
```
Comment (coocood) [inline sessionctx/stmtctx/stmtctx.go]: The number of backoff types is limited, the map would be small enough.

File: planner/core/planbuilder.go (hunk 1 above)
Comment (innerr) [reply planner/core/planbuilder.go]: Addressed

File: sessionctx/stmtctx/stmtctx.go (hunk 2 above)
Comment (innerr) [reply sessionctx/stmtctx/stmtctx.go]: Addressed

**Comment by coocood** [comment]
LGTM

**Pull request merged**